template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreConfigProvider.java

```diff
@@ -22,47 +22,36 @@
-        result.add(new ApiClusterTemplateConfig().name("hive_metastore_database_port").variable("hive-hive_metastore_database_port"));
-        result.add(new ApiClusterTemplateConfig().name("hive_metastore_database_host").variable("hive-hive_metastore_database_host"));
-        result.add(new ApiClusterTemplateConfig().name("hive_metastore_database_type").variable("hive-hive_metastore_database_type"));
-        result.add(new ApiClusterTemplateConfig().name("hive_metastore_database_name").variable("hive-hive_metastore_database_name"));
-        result.add(new ApiClusterTemplateConfig().name("hive_metastore_database_user").variable("hive-hive_metastore_database_user"));
-        return result;
-    }
-
-    @Override
-    public List<ApiClusterTemplateVariable> getServiceConfigVariables(TemplatePreparationObject source) {
-        List<ApiClusterTemplateVariable> result = new ArrayList<>();
-        Optional<RDSConfig> rdsConfigOptional = getFirstRDSConfigOptional(source);
+        Optional<RDSConfig> rdsConfigOptional = getFirstRDSConfigOptional(templatePreparationObject);
         Preconditions.checkArgument(rdsConfigOptional.isPresent());
         RdsView hiveView = new RdsView(rdsConfigOptional.get());
-        result.add(new ApiClusterTemplateVariable().name("hive-hive_metastore_database_host").value(hiveView.getHost()));
-        result.add(new ApiClusterTemplateVariable().name("hive-hive_metastore_database_port").value(hiveView.getPort()));
-        result.add(new ApiClusterTemplateVariable().name("hive-hive_metastore_database_name").value(hiveView.getDatabaseName()));
-        result.add(new ApiClusterTemplateVariable().name("hive-hive_metastore_database_type").value(hiveView.getSubprotocol()));
-        result.add(new ApiClusterTemplateVariable().name("hive-hive_metastore_database_password").value(hiveView.getConnectionPassword()));
-        result.add(new ApiClusterTemplateVariable().name("hive-hive_metastore_database_user").value(hiveView.getConnectionUserName()));
-        return result;
+        return List.of(
+                config("hive_metastore_database_host", hiveView.getHost()),
+                config("hive_metastore_database_name", hiveView.getDatabaseName()),
+                config("hive_metastore_database_password", hiveView.getConnectionPassword()),
+                config("hive_metastore_database_port", hiveView.getPort()),
+                config("hive_metastore_database_type", hiveView.getSubprotocol()),
+                config("hive_metastore_database_user", hiveView.getConnectionUserName())
+        );
     }
 
     @Override
     public String getServiceType() {
-        return "HIVE";
+        return HiveRoles.HIVE;
     }
 
     @Override
     public List<String> getRoleTypes() {
-        return Collections.singletonList("HIVEMETASTORE");
+        return List.of(HiveRoles.HIVEMETASTORE);
```

Comment:
Why this is better or worse than the singletonList?

Reply:
One less import.  Easier to add more elements or change to empty list.